Refactoring classes

Wikitelmex opened the pull request:
- script.js splited in 4 files
- classes made with SOLID principles aproach
- started styling

wikitelmex1 committed:
Update script.js

strict mode turned on

wikitelmex1 committed:
split file

- script.js will be splited in different files with different concerns

wikitelmex1 committed:
Update script.js

wikitelmex1 committed:
Update script.js

wikitelmex1 committed:
Update script.js

Commit:
form styled with responsiveness

Commit:
table data styled

Commit:
styles on a seperate css file

wikitelmex1 committed:
refactoring

- splited in 4 different files
- add array class
- add dom manipulation class
- add localstorage class
- some styling added

wikitelmex1 committed:
Linters fixed

wikitelmex1 committed:
Update script.js

wikitelmex1 committed:
Update index.html

DeliceLydia commented on script/array.js:
```diff
@@ -0,0 +1,58 @@
+}
+
+class BooksArray {
+  #books = []; // private encapsulation
+
+  AddBook(addTitle, addAuthor) {
+    const guindex = CreateGuid();
+    const book = {
+      index: guindex,
+      title: addTitle,
+      author: addAuthor,
+    };
+    this.#books.push(book);
+    return book;
+  }
+
```
- [x] I am not able to add a book in the books array please check your `addBook` method and fix where the issue is
![arre](https://user-images.githubusercontent.com/44745855/133495573-f54631bb-67a9-4f7e-ad96-571b2d410256.png)

This is the error am getting when I try to add a book

DeliceLydia replied:
- [x] It means if I can't add a book I can't test `remove` method please fix your methods

Wikitelmex replied:
was certain uncompatibility in some browsers, it was fixed

DeliceLydia reviewed (changes requested):
## Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

wikitelmex1 committed:
update array.js

- in certain browsers, loaded the localBooks as an boolean instead an array
- instead comparing to null before load, it compares to being an array

Davidosky007 commented:
## Hi, @Wikitelmex @peterdgreat, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 
![good-job-gif](https://user-images.githubusercontent.com/52497246/133646006-7d9380ba-69f4-4283-b724-562315f268cc.gif)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```
